aesara/tensor/slinalg.py

```diff
@@ -259,6 +259,152 @@ def cho_solve(c_and_lower, b, check_finite=True):
     return CholeskySolve(lower=lower, check_finite=check_finite)(A, b)
 
 
+class SolveTriangular(Op):
```

Comment:
Since a lot of code is duplicated between `Solve` and `SolveTriangular`, we should consolidate the logic and testing by creating an abstract base class&mdash;or by making `SolveTriangular` a subclass of `Solve`.

Reply:
Created new SolveBase class.